apps/src/templates/SchoolInfoConfirmationDialog.jsx

```diff
@@ -21,24 +21,45 @@ class SchoolInfoConfirmationDialog extends Component {
 
   constructor(props) {
     super(props);
-
     this.state = {
       showSchoolInterstitial: false,
       schoolName: props.schoolName,
       isOpen: true
     };
   }
 
+  static defaultProps = {
+    onUpdate: () => {},
+    schoolName: ''
+  };
+
+  componentDidMount() {
+    const {schoolName} = this.state;
+    if (!schoolName && schoolName.length > 0) {
+      fetch('/dashboardapi/v1/users/me/school_name')
+        .then(response => response.json())
+        .then(data => {
+          this.setState({
+            schoolName: data.school_name
+          });
+        })
+        .catch(error => this.setState({error}));
```

Comment:
I'm not familiar with this setState syntax - to what is the error assigned?

Reply:
If the request to fetch schoolName is unsuccessful, an error is thrown to activate the catch call.

Reply:
Sorry - my question should have been clearer! Once the catch call is activated, and `this.setState({error})` is called, to which part of the state is `error` set?

Reply:
Ah I see, it's the local state of the component
-